packages/data-stores/src/user/resolvers.ts

```diff
@@ -0,0 +1,18 @@
+/**
+ * Internal dependencies
+ */
+import { receiveCurrentUser, receiveCurrentUserFailed } from './actions';
+import { wpcomRequest } from '../utils';
+import { CurrentUser } from './types';
+
+export async function getCurrentUser() {
+	try {
+		const currentUser: CurrentUser = await wpcomRequest< CurrentUser >( {
+			path: '/me',
+			apiVersion: '1.1',
+		} );
+		return receiveCurrentUser( currentUser );
```

Comment:
I believe Gutenberg patters wouldn't actually make the request here, but would yield an action representing the request. _That_ action would trigger a side-effect via the corresponding `control` and when it resolved, the generator function would resume and and typically returns an action including the requested data to be populated in the store:

https://github.com/Automattic/wp-calypso/blob/b2a01ea5deb23f03daad6042c3a6d0e2bac53a92/packages/data-stores/src/verticals/resolvers.ts#L22-L29

Where the yielded action is:

```js
export const apiFetch = ( request ) => {
	return {
		type: 'API_FETCH',
		request,
	};
};
```

And there's a `control` that takes care of the side-effect:

```js
export const controls = {
	API_FETCH( { request } ) {
		return triggerFetch( request );
	},
```

(that's all from [`@wordpress/data-controls`](https://github.com/WordPress/gutenberg/blob/master/packages/data-controls/src/index.js#L123-L126))

I'm noting this as an observation and a pattern that we've been following because it's pattern core follows. However, the benefits of all the indirection still aren't entirely clear to me.

Reply:
Thanks, this makes sense to me — I've refactored to this kind of pattern.